crates/provider-blobstore-nats/src/provider.rs

```diff
@@ -0,0 +1,391 @@
+                let seed = Arc::new(seed);
+                async_nats::ConnectOptions::with_jwt(jwt, move |nonce| {
+                    let seed = seed.clone();
+                    async move { seed.sign(&nonce).map_err(async_nats::AuthError::new) }
+                })
+            }
+            (None, None) => async_nats::ConnectOptions::default(),
+            _ => bail!("must provide both jwt and seed for jwt authentication"),
+        };
+        if let Some(tls_ca) = &cfg.tls_ca {
+            opts = add_tls_ca(tls_ca, opts)?;
+        } else if let Some(tls_ca_file) = &cfg.tls_ca_file {
+            let ca = fs::read_to_string(tls_ca_file)
+                .await
+                .context("failed to read TLS CA file")?;
+            opts = add_tls_ca(&ca, opts)?;
+        }
+
+        // Get the cluster_uri with proper default
+        let uri = cfg.cluster_uri.unwrap_or(DEFAULT_NATS_URI.to_string());
+
+        // Connect to the NATS server
+        let client = opts
+            .name("NATS Object Store Provider")
+            .connect(uri.clone())
+            .await?;
+
+        // Get the JetStream context based on js_domain
+        let jetstream = if let Some(domain) = &cfg.js_domain {
+            async_nats::jetstream::with_domain(client.clone(), domain.clone())
+        } else {
+            async_nats::jetstream::new(client.clone())
+        };
+
+        debug!("opened NATS JetStream: {:?}", jetstream);
+        debug!("NATS Connection Configuration: {:?}", client);
+
+        // Return the handle to the opened NATS Object store
+        Ok(jetstream)
+    }
+
+    /// Helper function to lookup and return the NATS JetStream connection handle, and container storage
+    /// configuration, using the client component's context.
+    /// This ensures consistent implementation across all functions that need to get the NATS Blobstore.
+    pub(crate) async fn get_blobstore(
+        &self,
+        context: Option<Context>,
+    ) -> anyhow::Result<NatsBlobstore> {
+        if let Some((component_id, link_name)) =
+            context
+                .as_ref()
+                .and_then(|ctx @ Context { component, .. }| {
+                    component
+                        .clone()
+                        .map(|component_id| (component_id, ctx.link_name().to_string()))
+                })
+        {
+            // Acquire a read lock on the consumer components and attempt to find the specified component_id
+            let components = self.consumer_components.read().await;
+            let nats_stores = components
+                .get(&component_id)
+                .ok_or_else(|| anyhow!("consumer component not linked: {}", component_id))?;
+
+            // Get the NATS Object Store handle and its storage configuration
+            nats_stores
+                .get(&link_name)
+                .cloned()
+                .ok_or_else(|| anyhow!("no NATS Object Store found for link name: {}", &link_name))
```

Comment:
One more suggestion which is really just syntactic sugar since it aligns with our code style

```suggestion
                .ok_or_else(|| anyhow!("consumer component not linked: {component_id}"))?;

            // Get the NATS Object Store handle and its storage configuration
            nats_stores
                .get(&link_name)
                .cloned()
                .ok_or_else(|| anyhow!("no NATS Object Store found for link name: {link_name}"))
```